packages/dialog/src/DialogWrapper.ts

```diff
@@ -101,7 +101,7 @@ export class DialogWrapper extends DialogBase {
         return html`
             <sp-dialog
                 ?dismissable=${this.dismissable}
-                ?no-divider=${this.noDivider}
+                ?no-divider=${this.noDivider || !this.headline}
```

Comment:
This is much nicer than I thought it was going to have to be when I looked at this quickly the other week!